src/transformers/tokenization_utils_base.py

```diff
@@ -3252,3 +3191,113 @@ def _eventual_warn_about_too_long_sequence(self, ids: List[int], max_length: Opt
                     "indexing errors".format(len(ids), self.model_max_length)
                 )
             self.deprecation_warnings["sequence-length-is-longer-than-the-specified-maximum"] = True
+
+    @contextmanager
+    def as_target_tokenizer(self):
+        """
+        Temporarily sets the tokenizer for encoding the targets. Useful for tokenizer associated to
+        sequence-to-sequence models that need a slightly different processing for the labels.
+        """
+        yield
+
+    def prepare_seq2seq_batch(
```

Comment:
already add a deprecation warning now?

Reply:
Not yet, the idea was to have public examples (in the new seq2seq script) of *not* using this method before.